File: Java/Stack/Stack.java

class node
{
    public int data;
    public node next;

    public node(int no)
    {
        this.data = no;
        this.next = null;
    }
}

class StackX
{
    private node first;

    private int iCount;

    public StackX()
    {
        this.first = null;
        this.iCount = 0;
    }

    public void push(int no)
    {
        node newn = null;

        newn = new node(no);

        newn.next = this.first;
        this.first = newn;

        this.iCount++;
    }

    public int pop()
    {
        node temp = null;

        if(this.iCount == 0)
        {
            System.out.println("Stack is empty");
            return -1;
        }

        temp = this.first;
        int Value = first.data;

        this.first = this.first.next;
        System.gc();

        this.iCount--;
        return Value;
    }

    public int peep()
    {
        int Value = first.data;

        if(this.iCount == 0)
        {
            System.out.println("Stack is empty");
            return -1;
        }
        return Value;
    }

    public void Display()
    {
        node temp = null;
        if(this.iCount == 0)
        {
            System.out.println("Stack is empty\n");
            return;
        }

        temp =this.first;
        while(temp != null)
        {
            System.out.println("|   "+temp.data+"   |");
            temp = temp.next;
        }
    }

    public int Count()
    {
        return this.iCount;
    }
}

class Stack
{
    public static void main(String A[])
    {
        StackX sobj = new StackX();

        int iRet = 0;

        sobj.push(11);
        sobj.push(21);
        sobj.push(51);
        sobj.push(101);

        iRet = sobj.peep();
        System.out.println("Return value of peep is : "+iRet);

        sobj.Display();
        iRet = sobj.Count();
        System.out.println("Number of elements are : "+iRet);

        iRet = sobj.pop();
        System.out.println("Poped element is : "+iRet);

        sobj.Display();
        iRet = sobj.Count();
        System.out.println("Number of elements are : "+iRet);
    }

}

File: C/Stack/Stack.c
#include<stdio.h>
#include<stdlib.h>

#pragma pack(1)
struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void push(PPNODE first,int no)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = no;
    newn->next = NULL;

    newn->next = *first;
    *first = newn;
}

int pop(PPNODE first)
{
    int Value = (*first)->data;
    PNODE temp = NULL;

    temp = *first;

    if(first == NULL)
    {
        printf("Stck is empty\n");
    }

    *first = (*first)->next;
    free(temp);

    return Value;
}

int peep(PPNODE first)
{
    int Value = (*first)->data;

    if(first == NULL)
    {
        printf("Stck is empty\n");
    }

    return Value;
}

void Display(PNODE first)
{
    if(first == NULL)
    {
        printf("Stack is empty\n");
        return;
    }
    while(first != NULL)
    {
        printf("|  %d   |\n",first->data);
        first = first->next;
    }
}

int Count(PNODE first)
{
    int iCount = 0;
    while(first != NULL)
    {
        iCount++;
        first = first->next;
    }
    return iCount;
}

int main()
{
    PNODE head = NULL;
    int iRet = 0;

    push(&head,101);
    push(&head,51);
    push(&head,21);
    push(&head,11);

    iRet = peep(&head);
    printf("Return value of peep is : %d\n",iRet);

    Display(head);
    iRet = Count(head);
    printf("Number of elements in stack are : %d\n",iRet);

    iRet = pop(&head);

    printf("Poped element is : %d\n",iRet);

    Display(head);
    iRet = Count(head);
    printf("Number of elements in stack are : %d\n",iRet);

    return 0;
}